src/components/PrioritySelector/PrioritySelector.tsx

```diff
@@ -0,0 +1,147 @@
+import * as React from 'react';
+import Close from '@material-design-icons/svg/round/close.svg';
+import KeyboardDoubleArrowUp from '@material-design-icons/svg/round/keyboard_double_arrow_up.svg';
+import KeyboardArrowUp from '@material-design-icons/svg/round/keyboard_arrow_up.svg';
+import KeyboardArrowDown from '@material-design-icons/svg/round/keyboard_arrow_down.svg';
+import { DropdownMenuItem, Root, Portal } from '@radix-ui/react-dropdown-menu';
+import { DropdownTrigger } from '../Dropdown/DropdownTrigger';
+import { DropdownContent } from '../Dropdown/DropdownContent';
+import { bem } from '../../utils/bem';
+import { Text } from '../Text';
+import { IconButton } from '../Buttons';
+import itemStyles from '../Dropdown/Items/Item.scss';
+import styles from './PrioritySelector.scss';
+
+const iconMap = {
+    mandatory: KeyboardDoubleArrowUp,
+    important: KeyboardArrowUp,
+    optional: KeyboardArrowDown,
+    exclude: Close,
+};
+
+export enum Priority {
+    Mandatory = 'mandatory',
+    Important = 'important',
+    Optional = 'optional',
+    Exclude = 'exclude',
+}
+
+export type PriorityItemType<PriorityItemValue> = {
+    /** priority types: mandatory, important, optional or exclude. Makes correct icon show up */
+    priority: Priority;
+    /** text that should accompany the icon in the dropdown */
+    label: string;
+    /** optional: used in case of application-specific values / custom priorities */
+    value?: PriorityItemValue;
+};
+
+export interface Props<PriorityItemValue>
+    extends Omit<React.HTMLAttributes<HTMLDivElement>, 'onSelect'> {
+    /** Currently selected priority item that indicates the importance of the component. */
+    selectedItem: PriorityItemType<PriorityItemValue>;
+    /** Array of availible priority items. */
+    list: PriorityItemType<PriorityItemValue>[];
+    /** Callback function triggered when a new priority is selected. */
+    onSelect: (newPriorityItem: PriorityItemType<PriorityItemValue>) => void;
```

Comment:
What do you think about using `onChange`? It will be more consistent with `select` html and other libraries.

Reply:
https://www.radix-ui.com/primitives/docs/components/dropdown-menu radix uses onSelect instead of onChange for their DropdownMenuItem. That's why I used it here. I can change it to onChange, though, and call the onChange on onSelect from the DropdownMenuItem. Would that be preferable?

Reply:
Hmm, it also depends on how we use onSelect in OneUI. If it's common, we can keep it.

Reply:
It's not used loads. I changed it!